app/src/main/java/ahmaabdo/readify/rss/service/FetcherService.java

```diff
@@ -393,25 +393,23 @@ private void downloadAllImages() {
     }
 
     private void deleteOldEntries(long keepDateBorderTime) {
-        if (keepDateBorderTime > 0) {
-            String where = EntryColumns.DATE + '<' + keepDateBorderTime + Constants.DB_AND + EntryColumns.WHERE_NOT_FAVORITE;
-            // Delete the entries
-            MainApplication.getContext().getContentResolver().delete(EntryColumns.CONTENT_URI, where, null);
-            // Delete the cache files
-            NetworkUtils.deleteEntriesImagesCache(keepDateBorderTime);
-        }
         Cursor cursor = MainApplication.getContext().getContentResolver().query(FeedColumns.CONTENT_URI, new String[]{FeedColumns._ID, FeedColumns.KEEP_TIME}, null, null, null);
         while (cursor.moveToNext()) {
             long feedid = cursor.getLong(0);
             long keepTimeLocal = cursor.getLong(1) * 86400000l;
-            long keepDateBorderTimeLocal = keepTimeLocal > 0 ? System.currentTimeMillis() - keepTimeLocal : 0;
+            long keepDateBorderTimeLocal = keepTimeLocal > 0 ? System.currentTimeMillis() - keepTimeLocal : keepDateBorderTime;
             if (keepDateBorderTimeLocal > 0) {
                 String where = EntryColumns.DATE + '<' + keepDateBorderTimeLocal + Constants.DB_AND + EntryColumns.WHERE_NOT_FAVORITE + Constants.DB_AND + EntryColumns.FEED_ID + "=" + String.valueOf(feedid);
+                Cursor cursor2 = MainApplication.getContext().getContentResolver().query(EntryColumns.CONTENT_URI, EntryColumns.PROJECTION_ID, where, null, null);
+                while (cursor2.moveToNext()) {
+                    int entryId = cursor2.getInt(0);
+                    NetworkUtils.deleteEntryImagesCache(entryId, keepDateBorderTimeLocal);
+                }
+                cursor2.close();
```

Comment:
This part replaces lines [365](files#diff-92cfd6821d238f42e0d5321fa54b39d5L396) - 402, but respecting keep dates per feed.